### ISSUE TITLE
Norwegian Bokmål has "no" code in iD and presets imported from iD, Android expects "nb" code and fails to use iD preset data

### ISSUE THREAD

**Issue by matkoniecz** [streetcomplete/StreetComplete#3890]
https://poeditor.com/projects/po_edit?id=97843&term=273463027&id_language=127#showComments273463027

> The search field only accept english (for instance "bakery" instead of the Norwegian word "bakeri"

after this is fixed

> so I "translated" this to "A(n)… (på engelsk)", which means "A(n)... (in english)". A bit longer than the source, but I think it should be OK...

should be changed back/reset

**Comment by danieldegroot2** [comment]
> https://poeditor.com/projects/po_edit?id=97843&term=273463027&id_language=127#showComments273463027

Language id was incorrect, should be Norwegian Bokmal (NB); 1**1**7 > `quest_shop_gone_replaced_answer`
https://poeditor.com/projects/po_edit?id=97843&term=273463027&id_language=117#showComments273463027

**Comment by matkoniecz** [comment]
1) in currently deployed iD presets `shop=bakery` is shown with label Bakeri when preferred language is set to `no` code in OSM account setting/preferences (forgot [which one](https://github.com/openstreetmap/openstreetmap-website/issues/3167#issuecomment-1073073103))

2) when you set phone to Norsh bokmal then StreetComplete will not use Norwegian iD preset data - it will use English ones (shown in object descriptions, impacting "this object does not exist anymore" as reported here) 

3) When you set phone to Polish it works fine ("piekarnia" is found, presumable also in German)

---------------------------------

Norwegian may be a problem here - see https://en.wikipedia.org/wiki/Norwegian_language_conflict

There are at least two Norwegian languages https://en.wikipedia.org/wiki/Bokm%C3%A5l (nb code) and https://en.wikipedia.org/wiki/Nynorsk (nn code)

There are several possibilities, the most likely are: 

(1) Norwegian is not yet translated or was translated recently for iD presets, SC should reimport this data. Disproved by `rg bakeri` finding it in app/src/main/assets/osmfeatures/default/no.json

(2) Norwegian multilanguage mess is impacting SC

Bakeri is listed in `app/src/main/assets/osmfeatures/default/no.json` - note `no` language code used by iD (https://www.transifex.com/openstreetmap/id-editor/ has "Norwegian" and "Norwegian Nynorsk") which may be mismatching languages used by Android which has "Norwegian Bokmål" and "Norwegian Nynorsk"

**Comment by matkoniecz** [comment]
`app/src/main/assets/osmfeatures/default` has `no` and `nn` - maybe it should have `nb` and `nn`?

It fixes problem in Android version which I used for tests 

As bonus, it may depend on Android version, I am trying to check has handling of Norwegian changed in past...

**Comment by matkoniecz** [comment]
As browsers use `no` code I would not consider iD using it as a bug.

But using in Android is a bug.

I think that optimal solution is to add an extra step in https://github.com/streetcomplete/StreetComplete/blob/master/buildSrc/src/main/java/UpdatePresetsTask.kt that would move `no` file to `nb` file.

**Comment by matkoniecz** [comment]
Looking at: https://github.com/streetcomplete/StreetComplete/blob/6aca0dc9476034466d04949ffe870075932edb96/buildSrc/src/main/java/Utils.kt

It is possible that either `bcp47LanguageTagToJavaLanguageTag` should be changed or `javaLanguageTagToAndroidResCodes` used somehow. Note that there are also some other special cases, I tested and British English is working fine right now.

Serbian cyrillic is not listed at all in https://github.com/streetcomplete/StreetComplete/tree/master/app/src/main/assets/osmfeatures/default - but iD preset right now are not being translated into it at all (see https://www.transifex.com/openstreetmap/presets/dashboard/  - it has "request language" at the bottom)

I tried testing Chinese but due to fact that small part of iD presets is translated I was unable to find a good test case.
